src/groupNoble.ts

```diff
@@ -0,0 +1,338 @@
+    switch (gid) {
+        case GroupNb.ID.P256:
+            return { hash: sha256, L: 48 }
+        case GroupNb.ID.P384:
+            return { hash: sha384, L: 72 }
+        case GroupNb.ID.P521:
+            return { hash: sha512, L: 98 }
+        default:
+            throw errBadGroup(gid)
+    }
+}
+
+export class EltNb implements Elt {
+    private constructor(public readonly g: GroupNb, private readonly p: ProjPointType<bigint>) {}
+
+    static new(g: GroupNb): EltNb {
+        return new EltNb(g, getCurve(g.id).Point.ZERO)
+    }
+
+    static gen(g: GroupNb): EltNb {
+        return new EltNb(g, getCurve(g.id).Point.BASE)
+    }
+
+    isIdentity(): boolean {
+        return this.p.equals(getCurve(this.g.id).Point.ZERO)
+    }
+
+    isEqual(a: EltNb): boolean {
+        compat(this, a)
+        return this.p.equals(a.p)
+    }
+
+    neg(): EltNb {
+        return new EltNb(this.g, this.p.negate())
+    }
+
+    add(a: EltNb): EltNb {
+        compat(this, a)
+        return new EltNb(this.g, this.p.add(a.p))
+    }
+
+    mul(s: ScalarNb): EltNb {
+        compat(this, s)
+        return new EltNb(this.g, this.p.multiply(s.k))
+    }
+
+    mul2(k1: ScalarNb, a: EltNb, k2: ScalarNb): EltNb {
+        compat(this, k1)
+        compat(this, k2)
+        compat(this, a)
+        const el = this.p.multiplyAndAddUnsafe(a.p, k1.k, k2.k)
+        if (!el) throw new Error('result is zero')
+        return new EltNb(this.g, el)
+    }
+
+    serialize(compressed = true): Uint8Array {
+        if (this.p.equals(getCurve(this.g.id).Point.ZERO)) {
+            return Uint8Array.from([0])
+        }
+        return this.p.toRawBytes(compressed)
+    }
+
+    // size returns the number of bytes of a non-zero element in compressed or uncompressed form.
+    static size(g: GroupNb, compressed = true): number {
+        return 1 + (compressed ? g.size : g.size * 2)
+    }
+
+    private static deser(g: GroupNb, bytes: Uint8Array): EltNb {
+        const curve = getCurve(g.id)
+        const point = curve.Point.fromHex(bytes)
+        point.assertValidity()
+        return new EltNb(g, point)
+    }
+
+    // Deserializes an element, handles both compressed and uncompressed forms.
+    static deserialize(g: GroupNb, bytes: Uint8Array): EltNb {
+        const len = bytes.length
+        switch (true) {
+            case len === 1 && bytes[0] === 0x00:
+                return g.identity()
+            case len === 1 + g.size && (bytes[0] === 0x02 || bytes[0] === 0x03):
+                return EltNb.deser(g, bytes)
+            case len === 1 + 2 * g.size && bytes[0] === 0x04:
+                return EltNb.deser(g, bytes)
+            default:
+                throw errDeserialization(EltNb)
+        }
+    }
+
+    static hash(g: GroupNb, msg: Uint8Array, dst: Uint8Array): EltNb {
+        const h2c = getCurve(g.id).h2c
+        const DST = new TextDecoder().decode(dst)
+        const p = h2c(msg, { DST }) as ProjPointType<bigint>
```

Comment:
hash to curve accepts `Uint8Array`, no need to use `TextDecoder`

Reply:
Not sure if a bug, but it seemed to throw saying it required a string

Reply:
https://github.com/paulmillr/noble-curves/issues/57

Reply:
Fixed by https://github.com/paulmillr/noble-curves/pull/58